Document ResNet architecture tweak (#5977)
pytorch/vision: docs/source/models/resnet.rst

The ResNet model is based on the `Deep Residual Learning for Image Recognition
<https://arxiv.org/abs/1512.03385>`_ paper.

.. note::
    Bottleneck in torchvision places the stride for downsampling at 3x3 convolution(self.conv2)
    while original implementation places the stride at the first 1x1 convolution(self.conv1)

>     according to "Deep residual learning for image recognition"https://arxiv.org/abs/1512.03385.

Reviewer: This hyperlink will not render correctly. Given we already have a link to the paper few lines above, perhaps we can just say:
```suggestion
    according to the paper.
```

>     This variant is also known as ResNet V1.5 and improves accuracy according to

Reviewer: I don't think it's useful to place long string URLs in the documentation. How about:
```suggestion
    This variant improves the accuracy and it's known as `ResNet V1.5 <https://ngc.nvidia.com/catalog/model-scripts/nvidia:resnet_50_v1_5_for_pytorch>`_.
```
